site/tests/integration/login-flow.test.ts

```diff
@@ -0,0 +1,201 @@
+const accountDropdownButtonSelector = '[data-testid="account-dropdown-button"]';
+
+test("login works for an existing user (via verification code)", async ({
+  page,
+  isMobile,
+}) => {
+  await execa("yarn", ["exe", "scripts/seed-db.ts"]);
+
+  await page.goto("/");
+  const loginModal = await openLoginModal({ page, isMobile });
+
+  const emailInput = loginModal.locator(emailInputSelector);
+  const loginButton = loginModal.locator(loginButtonSelector);
+
+  await emailInput.fill("alice@example.com");
+  await loginButton.click();
+
+  await expect(loginModal.locator("text=Check your inbox")).toBeVisible();
+
+  const verificationCodeInput = loginModal.locator(
+    verificationCodeInputSelector,
+  );
+
+  await verificationCodeInput.fill(
+    await readValueFromRecentDummyEmail("Email verification code: "),
+  );
+  await verificationCodeInput.press("Enter");
+
+  const accountDropdownButton = page.locator(accountDropdownButtonSelector);
+  await expect(accountDropdownButton).toBeVisible();
+
+  await accountDropdownButton.click();
+  await expect(accountDropdownButton).toHaveText("A");
+
+  const accountDropdownPopover = page.locator(
+    '[data-testid="account-dropdown-popover"]',
+  );
+
+  await expect(accountDropdownPopover).toBeVisible();
+  await expect(accountDropdownPopover).toContainText("Alice@alice");
+
+  for (const [name, href] of [
+    ["Dashboard", "/dashboard"],
+    ["Your Profile", "/@alice"],
+    ["API Keys", "/settings/api-keys"],
+  ] as const) {
+    const item = accountDropdownPopover.locator(`a:has-text("${name}")`);
+    await expect(item).toBeVisible();
+    await expect(item).toHaveAttribute("href", href);
+  }
+
+  const logOutItem = accountDropdownPopover.locator(`span:has-text("Log Out")`);
+  await expect(logOutItem).toBeVisible();
+  await expect(logOutItem).toHaveAttribute("href", "");
+
+  await logOutItem.click();
+
+  await expect(accountDropdownButton).toBeHidden();
+
+  await openLoginModal({ page, isMobile });
+});
+
+test("login works for an existing user (via magic link)", async ({
+  browserName,
+  context,
+  isMobile,
+  page,
+}) => {
+  test.skip(
+    browserName === "webkit",
+    "https://app.asana.com/0/1202542409311090/1202652399221616",
+  );
+
+  await execa("yarn", ["exe", "scripts/seed-db.ts"]);
+
+  await page.goto("/");
+  const loginModal = await openLoginModal({ page, isMobile });
+
+  const emailInput = loginModal.locator(emailInputSelector);
+
+  await emailInput.fill("alice@example.com");
+  await emailInput.press("Enter");
+
+  await expect(loginModal.locator("text=Check your inbox")).toBeVisible();
+
+  const magicLink = await readValueFromRecentDummyEmail("Magic Link: ");
+
+  const page2 = await context.newPage();
+  await page2.goto(magicLink);
+
+  const accountDropdownButton2 = page2.locator(accountDropdownButtonSelector);
+  await expect(accountDropdownButton2).toBeVisible();
+  await expect(accountDropdownButton2).toHaveText("A");
```

Comment:
feels like we'd just need 1 of these... Can't have text "A" if it isn't visible. Can be addressed in a follow up

Reply:
`toBeVisible()` checks whether the element is actually visible on the screen as opposed to be just present in DOM:
https://playwright.dev/docs/actionability#visible

I understand that it's unlikely to pass `toHaveText` without `toBeVisible` but I'd still keep it here just in case.